test-case/test-mic-privacy.sh

```diff
@@ -0,0 +1,180 @@
+##
+## Description:
+##    Run alsabat process perform both playback and capture.
+##    Enable MIC privacy.
+##    Run alsabat process perform both playback and capture again.
+##
+## Case step:
+##    1. Specify the pcm IDs for playback and capture
+##    2. Check if usbrelay is installed and connected properly.
+##    3. Run alsabat process perform both playback and capture.
+##    4. Switch relay 1 to enable MIC privacy.
+##    5. Run alsabat process perform both playback and capture.
+##
+## Expect result:
+##    After step 3 the return value is 0.
+##    After step 5 the return value is -1001 (no peak be detected).
+
+TESTDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+TESTLIB="${TESTDIR}/case-lib"
+
+# remove the existing alsabat wav files
+ALSABAT_WAV_FILES="/tmp/mc.wav.*"
+rm -f "$ALSABAT_WAV_FILES"
+
+# shellcheck source=case-lib/lib.sh
+source "${TESTLIB}/lib.sh"
+source "${TESTLIB}/relay.sh"
+
+OPT_NAME['p']='pcm_p'     	OPT_DESC['p']='pcm for playback. Example: hw:0,0'
+OPT_HAS_ARG['p']=1          OPT_VAL['p']='hw:0,0'
+
+OPT_NAME['N']='channel_p'   OPT_DESC['N']='channel number for playback.'
+OPT_HAS_ARG['N']=1          OPT_VAL['N']='2'
+
+OPT_NAME['c']='pcm_c'      	OPT_DESC['c']='pcm for capture. Example: hw:0,1'
+OPT_HAS_ARG['c']=1          OPT_VAL['c']='hw:0,1'
+
+OPT_NAME['C']='channel_c'   OPT_DESC['C']='channel number for capture.'
+OPT_HAS_ARG['C']=1          OPT_VAL['C']='2'
+
+OPT_NAME['s']='sof-logger'  OPT_DESC['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_HAS_ARG['s']=0          OPT_VAL['s']=1
+
+OPT_NAME['r']='rate'        OPT_DESC['r']='sample rate'
+OPT_HAS_ARG['r']=1          OPT_VAL['r']=48000
+
+OPT_NAME['R']='relay'       OPT_DESC['R']='name of usbrelay switch, default value is HURTM_1'
+OPT_HAS_ARG['R']=1          OPT_VAL['R']='HURTM_1'
+
+func_opt_parse_option "$@"
+
+pcm_p=${OPT_VAL['p']}
+pcm_c=${OPT_VAL['c']}
+channel_c=${OPT_VAL['C']}
+channel_p=${OPT_VAL['N']}
+rate=${OPT_VAL['r']}
+relay=${OPT_VAL['R']}
+
+dlogi "Params: pcm_p=$pcm_p, pcm_c=$pcm_c, channel_c=$channel_c, channel_p=$channel_p, rate=$rate, LOG_ROOT=$LOG_ROOT"
+
+__upload_wav_files()
+{
+    # upload the alsabat wav file
+    for file in $ALSABAT_WAV_FILES; do
+        # alsabat has a bug where it creates an empty record in playback mode
+        if test -s "$file"; then
+            cp -v "$file" "$LOG_ROOT/"
+        fi
+    done
+}
+
+check_playback_capture()
+{
+    # check if capture and playback work
+    dlogc "alsabat -P$pcm_p -C$pcm_c -c 2 -r $rate"
+    alsabat -P"$pcm_p" -C"$pcm_c" -c 2 -r "$rate" || {
+        # upload failed wav file
+        __upload_wav_files
+        dle "alsabat failed"
+    }
+}
+
+main()
+{
+    setup_kernel_check_point
+
+    start_test
+
+    logger_disabled || func_lib_start_log_collect
+
+    if [ "$pcm_p" = "" ]||[ "$pcm_c" = "" ]; then
+        skip_test "No playback or capture PCM is specified. Skip the alsabat test."
```

Comment:
```suggestion
        skip_test "No playback or capture PCM is specified. Skip the $0 test."
```

There is already some test-case/alsabat.sh somewhere.